core/store/src/opener.rs

```diff
@@ -232,25 +229,22 @@ impl<'a> StoreOpener<'a> {
 
         if let Some(hot_meta) = hot_meta {
             if let Some(Some(cold_meta)) = cold_meta {
-                assert!(cfg!(feature = "cold_store"));
                 // If cold database exists, hot and cold databases must have the
-                // same version and to be Hot and Cold kinds respectively.
+                // same version and to be Hot and Cold or Archive and Cold kinds respectively.
```

Comment:
ditto, now we allow both archival and hot in the hot db. It stays archival for the duration of the migration and only changes to hot when we swap the db for a rpc converted to hot.